docs/polygon_draw.js

```diff
@@ -366,22 +387,79 @@ function setUpPolygonDrawing(map, firebasePromise) {
  * @param {google.maps.Map} map Map to display regions on
  * @param {Promise<any>} firebasePromise Promise with Firebase damage data (also
  *     implies that authentication is complete)
- * @return {Promise} promise that is resolved when all initialization is done
- *     (only used by tests)
+ * @return {?google.maps.drawing.DrawingManager} DrawingManager added to map
+ *     (only used by tests). Null if there was an error
  */
-function initializeAndProcessUserRegions(map, firebasePromise) {
+async function initializeAndProcessUserRegions(map, firebasePromise) {
   addLoadingElement(mapContainerId);
-  return firebasePromise
-      .then((doc) => {
-        // Damage asset may not exist yet, so this is undefined. We tolerate
-        // gracefully.
-        damageAsset = doc.data()['asset_data']['damage_asset_path'];
-        userShapes = getFirestoreRoot().collection(collectionName);
-      })
-      .then(() => userShapes.get())
-      .then(
-          (querySnapshot) => drawRegionsFromFirestoreQuery(querySnapshot, map))
-      .catch(createError('getting user-drawn regions'));
+  try {
+    setUpPopup();
+    let doc;
+    try {
+      doc = await firebasePromise;
+    } catch (err) {
+      // Firebase retrieval error handled elsewhere.
+      throw err;
+    }
+    // Damage asset may not exist yet, so this is undefined. We tolerate
+    // gracefully.
+    damageAsset = doc.data()['asset_data']['damage_asset_path'];
+    userShapes = getFirestoreRoot().collection(collectionName);
+    let querySnapshot;
+    try {
+      querySnapshot = await userShapes.get();
+    } catch (err) {
+      handleUserShapesError(err);
+      return null;
+    }
+    drawRegionsFromFirestoreQuery(querySnapshot, map);
+    return setUpPolygonDrawing(map);
+  } finally {
+    loadingElementFinished(mapContainerId);
+  }
+}
+
+const AUTHENTICATION_ERROR_CODE = 'permission-denied';
+const USER_FEATURES_DIALOG =
+    '<div>Sign in to authorized account to view user-drawn features</div>';
```

Comment:
maybe \<p\> instead?

Reply:
Is there a reason for \<p> over \<div?

Reply:
I think \<div> is better for a generic container. For instance, browsers add more space by default for \<p>
<img width="628" alt="Screen Shot 2019-12-26 at 4 44 22 PM" src="https://user-images.githubusercontent.com/10134896/71491609-977e4500-27ff-11ea-8a41-3fe3fe9225cd.png">
versus \<div 
<img width="624" alt="Screen Shot 2019-12-26 at 4 44 51 PM" src="https://user-images.githubusercontent.com/10134896/71491613-a1a04380-27ff-11ea-9cf7-010651a6bf2a.png">.
That's not actually what we want by swapping tags, and if we do want more space, setting it in the div is straightforward.